cpp/src/link_prediction/similarity_impl.cuh

```diff
@@ -58,7 +59,62 @@ rmm::device_uvector<weight_t> similarity(
     //        max weight((u,a), (a,v)).
     //    Use these to compute weighted score
     //
-    CUGRAPH_FAIL("weighted similarity computations are not supported in this release");
```

Comment:
Nothing to check if `do_expensive_check` flag is set? In multi-GPU, `vertex_pairs` should be properly distributed following the edge partitioning, right?